language/compiler/ir-to-bytecode/syntax/src/lexer.rs

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::syntax::ParseError;
+extern crate regex as __regex;
+use std::{convert::TryInto, fmt};
+
+// FIXME: This is a simplified version of the lexer generated by lalrpop.
+// It should be replaced with something sane.
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Tok {
+    EOF = -1,
+    AccountAddressValue = 0,
+    U64Value = 1,
+    NameValue = 2,
+    NameBeginTyValue = 3,
+    DotNameValue = 4,
+    ByteArrayValue = 5,
+    Exclaim = 6,
+    ExclaimEqual = 7,
+    Percent = 8,
+    Amp = 9,
+    AmpAmp = 10,
+    AmpMut = 11,
+    LParen = 12,
+    RParen = 13,
+    Star = 14,
+    Plus = 15,
+    Comma = 16,
+    Minus = 17,
+    Period = 18,
+    Slash = 19,
+    Colon = 20,
+    Semicolon = 21,
+    Less = 22,
+    LessEqual = 23,
+    Equal = 24,
+    EqualEqual = 25,
+    Greater = 26,
+    GreaterEqual = 27,
+    Caret = 28,
+    Underscore = 29,
+    Abort = 30,
+    Acquires = 31,
+    Address = 32,
+    As = 33,
+    Assert = 34,
+    Bool = 35,
+    BorrowGlobal = 36,
+    BorrowGlobalMut = 37,
+    Break = 38,
+    Bytearray = 39,
+    Continue = 40,
+    Copy = 41,
+    CreateAccount = 42,
+    Else = 43,
+    Exists = 44,
+    False = 45,
+    Freeze = 46,
+    GetGasRemaining = 47,
+    GetTxnGasUnitPrice = 48,
+    GetTxnMaxGasUnits = 49,
+    GetTxnPublicKey = 50,
+    GetTxnSender = 51,
+    GetTxnSequenceNumber = 52,
+    If = 53,
+    Import = 54,
+    Let = 55,
+    Loop = 56,
+    Main = 57,
+    Module = 58,
+    Modules = 59,
+    Move = 60,
+    MoveFrom = 61,
+    MoveToSender = 62,
+    Native = 63,
+    Public = 64,
+    Resource = 65,
+    Return = 66,
+    Script = 67,
+    Struct = 68,
+    True = 69,
+    U64 = 70,
+    Unrestricted = 71,
+    While = 72,
+    LBrace = 73,
+    Pipe = 74,
+    PipePipe = 75,
+    RBrace = 76,
```

Comment:
It might be nice if we aligned the `=` and then ignored rustfmt/cargo fmt 

It would make refactoring and renumbering a bit easier (at least for me)

Reply:
These numbers should go away. They're only there because the enum values have to align with the array of regexps from the lalrpop-generated code. I can clean up the lexer implementation next. I don't think it makes sense to mess with the formatting now, since it's just a temporary hack.